documentation/src/docs/state-synchronizer-design.md

```diff
@@ -7,24 +7,28 @@ You may obtain a copy of the License at
+This model works well when most of the updates are small in comparison to the total data size being stored, as they can be written as small deltas. As with any optimistic concurrency system it would work worst when many processes contend and try to update the same information at the same time.
 
 # Example
-A concrete example synchroning the contents of a set is [checked in here](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/SetSynchronizer.java). We also have an example that is synchronizing [membership of a set of hosts](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/MembershipSynchronizer.java).
 
-Imagine you want many processes to share a Map. This can be done by creating by a StateSynchronizer, which will help coordinate changes to the map. Each client has their own copy of the map in memory and can apply updates by passing a generator to the StateSynchronizer. Every time an update is attempted the updated is recorded to a segment. Updates will fail unless the Map passed into the update method is consistent with all of the updates that have been recorded to the segment. If this occurs the generator is called with the latest state to try again. Thus the order of updates is defined by the order in which they are written to the segment.
+A concrete [example](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/SetSynchronizer.java) of synchronizing the contents of a set is provided. We also have an example that is synchronizing [membership of a set of hosts](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/MembershipSynchronizer.java).
 
-# How this is implemented
-For this to work we use two features of the Pravega Segment Store Service.
-## Conditional append
-The append method can specify what offset the append is expected to be at. If the append will not go in, nothing should be done and a failure should be returned to the client.
+Imagine you want many processes to share a Map. This can be done by creating the State Synchronizer, it will aid in coordinating the changes to the Map. Each client has their own copy of the Map in memory and can apply updates by passing a generator to the State Synchronizer. Every time an update is attempted the updates are recorded to the Stream Segment. Updates are successful when the Map passed into the update method is consistent with all of the updates that have been recorded to the Stream Segment. If this occurs the generator is called with the latest state to try again. Thus the order of updates is defined by the order in which they are written to the Stream Segment.
```

Comment:
I have a doubt with this sentence: `Every time an update is attempted the updates are recorded to the Stream Segment.`
Any "update attempt" is stored in the Stream Segment, or only if the update is actually successful?

Reply:
@tkaitchuck  please respond to the above comment

Reply:
Raul is correct. It should read "Every time an update is made the update is recorded to the Stream Segment"

Reply:
@tkaitchuck Accepted suggestion